lib/sparse/src/index/search_context.rs

```diff
@@ -267,60 +256,71 @@ impl<'a> SearchContext<'a> {
-            match next_min_id_in_others {
-                Some(next_min_id) => {
-                    match next_min_id.cmp(&element.record_id) {
-                        Ordering::Equal => {
-                            // if the next min id in the other posting lists is the same as the current one,
-                            // we can't prune the current element as it needs to be scored properly across posting lists
-                            return false;
-                        }
-                        Ordering::Less => {
-                            // we can't prune as there the other posting lists contains smaller smaller ids that need to scored first
-                            return false;
-                        }
-                        Ordering::Greater => {
-                            // next_min_id is > element.record_id there is a chance to prune up to `next_min_id`
-                            // check against the max possible score using the `max_next_weight`
-                            // we can under prune as we should actually check the best score up to `next_min_id` - 1 only
-                            // instead of the max possible score but it is not possible to know the best score up to `next_min_id` - 1
-                            let max_weight_from_list = element.weight.max(element.max_next_weight);
-                            let max_score_contribution =
-                                max_weight_from_list * longest_posting_iterator.query_weight;
-                            if max_score_contribution <= min_score {
-                                // prune to next_min_id
-                                let longest_posting_iterator =
-                                    &mut self.postings_iterators[0].posting_list_iterator;
-                                let position_before_pruning =
-                                    longest_posting_iterator.current_index;
-                                longest_posting_iterator.skip_to(next_min_id);
-                                let position_after_pruning = longest_posting_iterator.current_index;
-                                // check if pruning took place
-                                return position_before_pruning != position_after_pruning;
-                            }
-                        }
-                    }
+    /// Prune posting lists that cannot possibly contribute to the top results.
+    /// If the min record id is unique, we can prune up to the next min record id.
+    ///
+    /// Returns true if the a posting list was pruned.
+    pub fn prune_lowest_posting_list(&mut self, min_score: f32) -> bool {
+        let trackers_len = self.min_record_ids.len();
+        // no trackers left
+        if trackers_len == 0 {
+            return false;
+        }
+        // one tracker left, let's try to prune fully the corresponding posting
+        if trackers_len == 1 {
```

Comment:
Though perfectly fine, maybe we can do this  to prevent if-elsing:

```rust
match self.min_record_ids.len() {
    0 => return false,
    1 => {
        // ...
    }
    2.. => {
        // ...
    }
}
```